contracts/src/challengeV2/libraries/EdgeChallengeManagerLib.sol

```diff
@@ -71,6 +71,8 @@ struct EdgeStore {
     ///         When only one edge exists for a specific mutual id then a special magic string hash is stored instead
     ///         of the first rival id, to signify that a single edge does exist with this mutual id
     mapping(bytes32 => bytes32) firstRivals;
+    /// @notice A mapping of mutualId to the edge id of the confirmed rival with that mutualId
+    mapping(bytes32 => bytes32) rivalConfirmed;
```

Comment:
This add storage, we can also store a flag in the `firstRivals` edge but that is a bit ugly as the flag might not be directly relevant to the edge (e.g. the confirmed rival is not the one stored in firstRivals[mutualId])

Reply:
As discussed, we may also store firstRivals[mutualid] = SOME_MAGIC_BYTES32 to indicate the mutual id is confirmed already, need to further evaluate if that would break anything